pallets/services-payment/src/lib.rs

```diff
@@ -249,6 +257,30 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        /// Max core price for parathread in relay chain currency
+        #[pallet::call_index(5)]
+        #[pallet::weight(T::WeightInfo::set_max_core_price())]
+        pub fn set_max_core_price(
+            origin: OriginFor<T>,
+            para_id: ParaId,
+            max_core_price: Option<u128>,
+        ) -> DispatchResultWithPostInfo {
+            T::SetRefundAddressOrigin::ensure_origin(origin, &para_id)?;
```

Comment:
Might be good to rename `SetRefundAddressOrigin` to `ManagerOrigin` or similar if it's going to be used in multiple calls

Reply:
Yeah I would do that